Firmware/src/modules/tools/temperaturecontrol/TemperatureControl.cpp

```diff
@@ -512,6 +515,8 @@ float TemperatureControl::get_temperature()
 // called in an ISR, be nice!
 void TemperatureControl::thermistor_read_tick()
 {
+	// we schedule a call back in command context to get raw data if max31855 is configured
+    want_command_ctx = true;
```

Comment:
why is this enabled every time?

Reply:
This should be used only if a max31855 sensor is configured, i think it is missing just a if condition to check whether it is a max31855 or not, my suggestion would be adding if (this->sensor_type=="max31855") for this variable and the other from configure method, do you agree?

Reply:
Well that would be the correct way to do it except the whole idea of using the commnd_ctx callback is invalid in this case.